### PR TITLE
ci: update `.releaserc` to publish from appropriate branches

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
## Description

see https://github.com/openedx/paragon/issues/3272

Include a description of your changes here, along with a link to any relevant Jira tickets and/or Github issues.

### Deploy Preview

Include a direct link to your changes in this PR's deploy preview here (e.g., a specific component page).

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please request an a11y review for the PR in the [#wg-paragon](https://openedx.slack.com/archives/C02NR285KV4) Open edX Slack channel.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@openedx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: .releaserc [hunk 1]
```diff
@@ -1,9 +1,8 @@
 {
   "branches": [
-    "master",
-    "next",
-    { "name": "alpha", "prerelease": true },
-    { "name": "beta", "prerelease": true }
+    { "name": "release-22.x", "range": "22.x", "channel": "22.x" },
+    { "name": "release-23.x", "range": "23.x" },
```
Comment (brian-smith-tcril) [inline .releaserc:4]: if I understand the `channel` docs (https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#channel) correctly this should make it so the `release-23.x` branch publishes to the `@latest` tag on npm

File: .releaserc [hunk 2]
```diff
@@ -1,9 +1,8 @@
...
+    { "name": "next", "prerelease": true, "channel": "next" }
```
Comment (brian-smith-tcril) [inline .releaserc:5]: I'm not quite sure how `prerelease` and `channel` interact, but of all things to "test in prod" about this config it feels the least scary.

File: .releaserc [hunk 3]
```diff
@@ -1,9 +1,8 @@
 {
   "branches": [
-    "master",
-    "next",
-    { "name": "alpha", "prerelease": true },
-    { "name": "beta", "prerelease": true }
+    { "name": "release-22.x", "range": "22.x", "channel": "22.x" },
```
Comment (brian-smith-tcril) [inline .releaserc:3]: Based on https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration, this should make it so
* `name`: we're looking at the `release-22.x` branch
* `range`: this only publishes releases in the `22.x` range of semantic versions
* `channel`: these releases are published to the `22.x` tag on `npm`

**Review by adamstankiewicz — APPROVED** [review]

**Comment by brian-smith-tcril** [comment]
closing in favor of https://github.com/openedx/paragon/pull/3341

**Pull request closed without merging**